### ISSUE TITLE
Refactoring and Search Functionality for Open Sustainable Technology 

### ISSUE THREAD

**Issue by Ly0n** [protontypes/open-sustainable-technology#105]
The current design of the website makes it difficult to search and find projects according to criteria such as "programming languages" or specific topics. Therefore, the project always receives poor feedback on its usability. Therefore, we are looking for a concept to better integrate the previously created database into the existing website or to transfer it into a new website. 

The requirements are:
* Easy maintainability 
* Low energy consumption of the website
* Hosting on GitHub pages if possible
* As far as the functionality is not restricted, based on open source tools.

Database:
https://airtable.com/shr9we419r2TkpLkc
 
The aim of the website is to give beginners as well as experts in the field of sustainable technologies a possibility to discover software tools, projects, datasets, communities, tutorials and other resources easily related to their own problems and questions. 

First, a proof-of-concept is needed that shows how the website in its current state can be transferred to the new state.

**Comment by Ly0n** [comment]
@jtorreggiani As promised, here is the issue on the redesign of the website.

**Comment by NathanBaulch** [comment]
> according to criteria such as "programming languages"

Yes please, this was the very first thing I wanted to do after stumbling on this project. I'm open to learning new languages but would ideally hit the ground running with a language I already know.

**Comment by Ly0n** [comment]
> > according to criteria such as "programming languages"
> 
> Yes please, this was the very first thing I wanted to do after stumbling on this project. I'm open to learning new languages but would ideally hit the ground running with a language I already know.

At the moment you can use the AirTable to sort for "programming languages" . So far, however, we have not found anyone who integrates a search function for the database into the website.

**Comment by rjbergerud** [comment]
Airtable offers both a filter and a search functionality.

Any reason to not just embed the airtable?  Or were you thinking the interface isn't intuitive enough?

**Comment by Ly0n** [comment]
Thanks, @rjbergerud, for bringing this topic up again. You are absolutely right that the AirTable comes with all the search features one needs. In the past I had always hoped that we could replace AirTable an open source alternative like Datasette or NocoDB. Unfortunately, these variants always have to be hosted by yourself and so far I haven't had the time to look into them. Until we have found a better alternative, it makes sense to integrate the AirTable variant better. Here is the current integration with an iframe: https://opensustain.tech/discover/. What do you think about it? 

Unfortunately I don't manage to fill the whole tab of the website. Any idea how to extend the iframe to the whole tab?

**Comment by Ly0n** [comment]
There is one more problem with AirTable. A "base" has to be updated manually every time. AirTable does not inherently offer automatic synchronization with a CSV URL. The third-party applications I have found so far do not work reliably. The lack of such standard features is typical for closed source projects.
